src/app/clusters/commissioner-control-server/commissioner-control-server.cpp

```diff
@@ -0,0 +1,277 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "commissioner-control-server.h"
+
+#include <protocols/interaction_model/StatusCode.h>
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/cluster-enums.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/AttributeAccessInterfaceRegistry.h>
+#include <app/CommandHandler.h>
+#include <app/EventLogging.h>
+#include <app/reporting/reporting.h>
+#include <platform/PlatformManager.h>
+
+#include <memory>
+
+using namespace chip;
+using namespace chip::app;
+
+using chip::Protocols::InteractionModel::Status;
+
+namespace {
+
+NodeId getNodeId(const app::CommandHandler * commandObj)
+{
+    if (nullptr == commandObj || nullptr == commandObj->GetExchangeContext())
+    {
+        ChipLogError(Zcl, "Cannot access ExchangeContext of Command Object for Node ID");
+        return kUndefinedNodeId;
+    }
+
+    if (!commandObj->GetExchangeContext()->HasSessionHandle())
+    {
+        ChipLogError(Zcl, "Cannot access session of Command Object for Node ID");
+        return kUndefinedNodeId;
+    }
+
+    auto descriptor = commandObj->GetExchangeContext()->GetSessionHandle()->GetSubjectDescriptor();
```

Comment:
This is a really complicated way of writing:
```suggestion
    auto descriptor = commandObj->GetSubjectDescriptor();
```
Note that commandObj can't be null here either.